---
title: "Mouse RNA heatmap from human PC"
author: "Bing Shui"
date: "4/6/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Index 
* [ID conversion from human gene symbol to mouse Ensembl ID](#idconversion)
* [Heatmap plotting based on cCD PC](#cCDpc)
* [Heatmap plotting based on iCD PC](#iCDpc)
* [Heatmap plotting based on UC PC using MK2i dataset](#UCpc)

## Library setup
```{r fig.height = 10, fig.align = "center"}
suppressMessages(
  suppressWarnings(
    c(library(data.table),
      library(gdata),
      library(ggplot2),
      library(RColorBrewer),
      library(readr),
      library(tidyverse),
      library(plotly),
      library(directlabels),
      library(gplots),
      library(knitr)
      )))
```

## ID conversion from human gene symbol to mouse Ensembl ID {#idconversion}
This conversion is done using ID database from MGI to be consistent with Doug Brubaker's analysis using human RNA-Seq dataset. 

Import the dataset containing Gene ID information from MGI
```{r fig.height = 10, fig.align = "center"}
homolog_id <- read_csv("Human_2_mouse_symbol.csv")

ensembl_id <- read_csv("mouse_symbol_Ensembl.csv")
```
```{r fig.height = 10, fig.align = "center"}
# load the PC dataset
cCD_PC <- read_csv("cCD PC.csv")
iCD_PC <- read_csv("iCD PC.csv")

human_id <- as.data.frame(cCD_PC$Row)
human_id <- as_tibble(human_id)
colnames(human_id) <- "human_gene_symbol"

id_conversion <- inner_join(human_id, homolog_id, by = c("human_gene_symbol" = "Human Marker Symbol"))
non_duplicates_idx <- which(duplicated(id_conversion$human_gene_symbol) == FALSE)
id_conversion <- id_conversion[non_duplicates_idx, ]

id_conversion <- inner_join(id_conversion, ensembl_id, by = c("Mouse Marker Symbol" = "Marker Symbol"))
non_duplicates_idx <- which(duplicated(id_conversion$`Mouse Marker Symbol`) == FALSE)
id_conversion <- id_conversion[non_duplicates_idx, ]

id_conversion <- id_conversion[,c(1,7)]

cCD_PC <- inner_join(cCD_PC, id_conversion, by = c("Row" = "human_gene_symbol"))
iCD_PC <- inner_join(iCD_PC, id_conversion, by = c("Row" = "human_gene_symbol"))
```

## Heatmap plotting based on cCD PC {#cCDpc}
For cCD PCs, PC1 is selected because it seperates diseases statuses. PC10 and PC14 are selected because they seperate TNF from TCT mice.

The goal of this is to examine how many of the genes that contribute to the seperation of TNF and TCT model based on human PCs are targeted by MK2 inhibitor.

For each PC, the top 325 contributing genes are selected for plotting the heatmap (10% of the list). Top contributing genes are genes with the largest absolute loading coefficients.
```{r fig.height = 10, fig.align = "center"}
n = 10
threshold_p1 <- quantile(abs(cCD_PC$cCD_PC1), prob=1-n/100)
threshold_p10 <- quantile(abs(cCD_PC$cCD_PC10), prob=1-n/100)
threshold_p14 <- quantile(abs(cCD_PC$cCD_PC14), prob=1-n/100)

ensembl_p1 <- cCD_PC$`Ensembl Gene ID`[abs(cCD_PC$cCD_PC1) > threshold_p1]
ensembl_p10 <- cCD_PC$`Ensembl Gene ID`[abs(cCD_PC$cCD_PC10) > threshold_p10]
ensembl_p14 <- cCD_PC$`Ensembl Gene ID`[abs(cCD_PC$cCD_PC14) > threshold_p14]
```

### cCD_PC1
I will use I_V vs NI_V data to plot this heatmap since this is the PC that separates diseases status.

```{r fig.height = 10, fig.align = "center"}
TCT_iv_niv_vst <- read_csv("~/OneDrive - Harvard University/Haigis Lab/Projects/Colloboration Projects/IBD RNA-Seq analysis/Analysis/TCT/VST_I_V vs NI_V.csv")
TNF_iv_niv_vst <- read_csv("~/OneDrive - Harvard University/Haigis Lab/Projects/Colloboration Projects/IBD RNA-Seq analysis/Analysis/TNF/VST_I_V vs NI_V.csv")
TCT_iv_niv_dif <- read_csv("~/OneDrive - Harvard University/Haigis Lab/Projects/Colloboration Projects/IBD RNA-Seq analysis/Analysis/TCT/Differential Analysis_I_V vs NI_V.csv")
TNF_iv_niv_dif <- read_csv("~/OneDrive - Harvard University/Haigis Lab/Projects/Colloboration Projects/IBD RNA-Seq analysis/Analysis/TNF/Differential Analysis_I_V vs NI_V.csv")
```

#### TCT model
All genes in the list of 325 from PC1
```{r fig.height = 10, fig.align = "center"}
suppressMessages(library(mosaic))

pc1_genes <- TCT_iv_niv_vst[TCT_iv_niv_vst$X1 %in% ensembl_p1,]
pc1_genes <- as.data.frame(pc1_genes)

for (i in 1:dim(pc1_genes)[1]) {
  pc1_genes[i,2:12] <- zscore(as.numeric(pc1_genes[i,2:12]))
}

my_palette <- colorRampPalette(c("blue", "white", "red"))(128)
heatmap_matrix <- as.matrix(pc1_genes[,2:12])

png('TCT_I_V vs NI_V in cCD PC1 Genes.png',
    width = 600,
    height = 1200,
    res = 200,
    pointsize = 8)

heatmap.2(heatmap_matrix,
          main = "TCT I_V vs NI_V\nin cCD PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TCT_I_V vs NI_V in cCD PC1 Genes.pdf',
    width = 6,
    height = 10)

heatmap.2(heatmap_matrix,
          main = "TCT I_V vs NI_V\nin cCD PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics("TCT_I_V vs NI_V in cCD PC1 Genes.png")

```

DEG from PC1
```{r fig.height = 10, fig.align = "center"}
pc1_dif <- TCT_iv_niv_dif[TCT_iv_niv_dif$X1 %in% ensembl_p1,]
pc1_DEG <- subset(pc1_dif, pc1_dif$padj < 0.05  & abs(pc1_dif$log2FoldChange) > 0.25)
pc1_genes_DEG <- pc1_genes[pc1_genes$X1 %in% pc1_DEG$X1,]
pc1_genes_DEG <- inner_join(pc1_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))


heatmap_matrix <- as.matrix(pc1_genes_DEG[,2:12])

png('TCT_I_V vs NI_V DEG in cCD PC1 Genes.png',
    width = 800,
    height = 600,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix,
          main = "TCT_I_V vs NI_V DEG\nin cCD PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,3),
          col=my_palette,
          cexCol = 1,
          margins = c(4,6),
          trace = "none",
          dendrogram = "row",
          labRow = pc1_genes_DEG$`Marker Symbol`,
          keysize = 2,
          Colv = "NA")
dev.off()


pdf('TCT_I_V vs NI_V DEG in cCD PC1 Genes.pdf',
    width = 8,
    height = 6)
heatmap.2(heatmap_matrix,
          main = "TCT_I_V vs NI_V DEG\nin cCD PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,3),
          col=my_palette,
          cexCol = 1,
          margins = c(4,6),
          trace = "none",
          dendrogram = "row",
          labRow = pc1_genes_DEG$`Marker Symbol`,
          keysize = 2,
          Colv = "NA")
dev.off()

include_graphics("TCT_I_V vs NI_V DEG in cCD PC1 Genes.png")

```

6/324 genes are deemed significant.


#### TNF model
All genes in the list of 325 from PC1
```{r fig.height = 10, fig.align = "center"}

pc1_genes <- TNF_iv_niv_vst[TNF_iv_niv_vst$X1 %in% ensembl_p1,]
pc1_genes <- as.data.frame(pc1_genes)


for (i in 1:dim(pc1_genes)[1]) {
  pc1_genes[i,2:11] <- zscore(as.numeric(pc1_genes[i,2:11]))
}

heatmap_matrix <- as.matrix(pc1_genes[,2:11])

png('TNF_I_V vs NI_V in cCD PC1 Genes.png',
    width = 600,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix,
          main = "TNF I_V vs NI_V\nin cCD PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()


pdf('TNF_I_V vs NI_V in cCD PC1 Genes.pdf',
    width = 6,
    height = 10)
heatmap.2(heatmap_matrix,
          main = "TNF I_V vs NI_V\nin cCD PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics("TNF_I_V vs NI_V in cCD PC1 Genes.png")

```

DEG from PC1
```{r fig.height = 10, fig.align = "center"}
pc1_dif <- TNF_iv_niv_dif[TNF_iv_niv_dif$X1 %in% ensembl_p1,]
pc1_DEG <- subset(pc1_dif, pc1_dif$padj < 0.05 & abs(pc1_dif$log2FoldChange) > 0.25)
pc1_genes_DEG <- pc1_genes[pc1_genes$X1 %in% pc1_DEG$X1,]
pc1_genes_DEG <- inner_join(pc1_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))


heatmap_matrix <- as.matrix(pc1_genes_DEG[,2:11])

png('TNF_I_V vs NI_V DEG in cCD PC1 Genes.png',
    width = 600,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix,
          main = "TNF\nI_V vs NI_V DEG\ncCD PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(4,6),
          trace = "none",
          dendrogram = "row",
          labRow = pc1_genes_DEG$`Marker Symbol`,
          keysize = 2,
          Colv = "NA")
dev.off()


pdf('TNF_I_V vs NI_V DEG in cCD PC1 Genes.pdf',
    width = 6,
    height = 10)
heatmap.2(heatmap_matrix,
          main = "TNF\nI_V vs NI_V DEG\ncCD PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(4,6),
          trace = "none",
          dendrogram = "row",
          labRow = pc1_genes_DEG$`Marker Symbol`,
          keysize = 2,
          Colv = "NA")
dev.off()

include_graphics('TNF_I_V vs NI_V DEG in cCD PC1 Genes.png')

```
74/324 genes are deemed significant.

### cCD_PC10 and cCD_PC14
I_MKI vs I_V datasets will be used here. 

```{r fig.height = 10, fig.align = "center"}
TCT_imki_iv_vst <- read_csv("~/OneDrive - Harvard University/Haigis Lab/Projects/Colloboration Projects/IBD RNA-Seq analysis/Analysis/TCT/VST_I_MKI vs I_V.csv")
TCT_nimki_niv_vst <- read_csv("~/OneDrive - Harvard University/Haigis Lab/Projects/Colloboration Projects/IBD RNA-Seq analysis/Analysis/TCT/VST_NI_MKI vs NI_V.csv")
TCT_combined_vst <- read_csv("~/OneDrive - Harvard University/Haigis Lab/Projects/Colloboration Projects/IBD RNA-Seq analysis/Analysis/TCT/VST_MKI vs V.csv")
TNF_imki_iv_vst <- read_csv("~/OneDrive - Harvard University/Haigis Lab/Projects/Colloboration Projects/IBD RNA-Seq analysis/Analysis/TNF/VST_I_MKI vs I_V.csv")
TNF_nimki_niv_vst <- read_csv("~/OneDrive - Harvard University/Haigis Lab/Projects/Colloboration Projects/IBD RNA-Seq analysis/Analysis/TNF/VST_NI_MKI vs NI_V.csv")
TNF_combined_vst <- read_csv("~/OneDrive - Harvard University/Haigis Lab/Projects/Colloboration Projects/IBD RNA-Seq analysis/Analysis/TNF/VST_MKI vs V.csv")

TCT_imki_iv_dif <- read_csv("~/OneDrive - Harvard University/Haigis Lab/Projects/Colloboration Projects/IBD RNA-Seq analysis/Analysis/TCT/Differential Analysis_I_MKI vs I_V.csv")
TCT_nimki_niv_dif <- read_csv("~/OneDrive - Harvard University/Haigis Lab/Projects/Colloboration Projects/IBD RNA-Seq analysis/Analysis/TCT/Differential Analysis_NI_MKI vs NI_V.csv")
TCT_combined_dif <- read_csv("~/OneDrive - Harvard University/Haigis Lab/Projects/Colloboration Projects/IBD RNA-Seq analysis/Analysis/TCT/Differential Analysis_MKI vs V.csv")
TNF_imki_iv_dif <- read_csv("~/OneDrive - Harvard University/Haigis Lab/Projects/Colloboration Projects/IBD RNA-Seq analysis/Analysis/TNF/Differential Analysis_I_MKI vs I_V.csv")
TNF_nimki_niv_dif <- read_csv("~/OneDrive - Harvard University/Haigis Lab/Projects/Colloboration Projects/IBD RNA-Seq analysis/Analysis/TNF/Differential Analysis_NI_MKI vs NI_V.csv")
TNF_combined_dif <- read_csv("~/OneDrive - Harvard University/Haigis Lab/Projects/Colloboration Projects/IBD RNA-Seq analysis/Analysis/TNF/Differential Analysis_MKI vs V.csv")
```

#### TCT model stratifying based on inflammation status
All genes in the list of 325 from PC10 and PC14
```{r fig.height = 10, fig.align = "center"}

pc10_genes <- as.data.frame(TCT_imki_iv_vst[TCT_imki_iv_vst$X1 %in% ensembl_p10,])
pc14_genes <- as.data.frame(TCT_imki_iv_vst[TCT_imki_iv_vst$X1 %in% ensembl_p14,])


for (i in 1:dim(pc10_genes)[1]) {
  pc10_genes[i,2:13] <- zscore(as.numeric(pc10_genes[i,2:13]))
}

for (i in 1:dim(pc14_genes)[1]) {
  pc14_genes[i,2:13] <- zscore(as.numeric(pc14_genes[i,2:13]))
}

heatmap_matrix_pc10 <- as.matrix(pc10_genes[,2:13])
heatmap_matrix_pc14 <- as.matrix(pc14_genes[,2:13])


png('TCT_I_MKI vs I_V in cCD PC10 Genes.png',
    width = 600,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc10,
          main = "TCT I_MKI vs I_V\nin cCD PC10 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TCT_I_MKI vs I_V in cCD PC10 Genes.pdf',
    width = 6,
    height = 10)
heatmap.2(heatmap_matrix_pc10,
          main = "TCT I_MKI vs I_V\nin cCD PC10 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TCT_I_MKI vs I_V in cCD PC10 Genes.png')

png('TCT_I_MKI vs I_V in cCD PC14 Genes.png',
    width = 600,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc14,
          main = "TCT I_MKI vs I_V\nin cCD PC14 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TCT_I_MKI vs I_V in cCD PC14 Genes.pdf',
    width = 6,
    height = 10)
heatmap.2(heatmap_matrix_pc14,
          main = "TCT I_MKI vs I_V\nin cCD PC14 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TCT_I_MKI vs I_V in cCD PC14 Genes.png')
```


DEG from PC10 and PC14. The standard of DEG is relaxed to p < 0.1 here to keep consistent with previous analysis and LFC > 0.1.
```{r fig.height = 10, fig.align = "center"}
pc10_dif <- TCT_imki_iv_dif[TCT_imki_iv_dif$X1 %in% ensembl_p10,]
pc14_dif <- TCT_imki_iv_dif[TCT_imki_iv_dif$X1 %in% ensembl_p14,]

pc10_DEG <- subset(pc10_dif, pc10_dif$padj < 0.1  & abs(pc10_dif$log2FoldChange > 0.1))
pc14_DEG <- subset(pc14_dif, pc14_dif$padj < 0.1  & abs(pc14_dif$log2FoldChange > 0.1))

pc10_genes_DEG <- pc10_genes[pc10_genes$X1 %in% pc10_DEG$X1,]
pc14_genes_DEG <- pc14_genes[pc14_genes$X1 %in% pc14_DEG$X1,]

pc10_genes_DEG <- inner_join(pc10_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))
pc14_genes_DEG <- inner_join(pc14_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))

print(paste("The number of MK2 inhibitor targets from TCT model from cCD PC10:", dim(pc10_genes_DEG)[1]))
print(paste("The number of MK2 inhibitor targets from TCT model from cCD PC14:", dim(pc14_genes_DEG)[1]))
```

We cannot make heatmap with only 1 item as input. So I will just print out the item here.
```{r fig.height = 10, fig.align = "center"}
pc14_DEG
```

I would also like to plot the Loading plot for this gene in PC10 and PC14 space.

```{r}
loading_cord <- cCD_PC[cCD_PC$`Ensembl Gene ID` %in% pc14_genes_DEG$X1,]
loading_cord <- loading_cord[, c('cCD_PC10', 'cCD_PC14', 'Ensembl Gene ID')]
loading_cord <- inner_join(loading_cord, ensembl_id[,c(2,4)], by = c("Ensembl Gene ID"="Ensembl Gene ID"))
loading_cord$`Ensembl Gene ID` = NULL
loading_cord[2,] <- loading_cord[1,]
loading_cord <- as.data.frame(loading_cord)
loading_cord[1,] <- c(0,0,NA)
loading_cord$cCD_PC10 <- as.numeric(loading_cord$cCD_PC10)
loading_cord$cCD_PC14 <- as.numeric(loading_cord$cCD_PC14)


ggplot(loading_cord, aes(x=cCD_PC14, y=cCD_PC10)) + 
  geom_line(color = "red",
            arrow = arrow()) +
  xlim(-0.1, 0.1) +
  ylim(-0.1, 0.1) +
  ggtitle("Loading plot for TCT MK2i target in cCD PC10 and PC14 space") +
  geom_dl(aes(label = loading_cord$`Marker Symbol`), method = list(dl.trans(x = x + 0.2), "last.points", cex = 0.8))

```

#### TCT model disregarding inflammation status
All genes in the list of 325 from PC10 and PC14
```{r fig.height = 10, fig.align = "center"}

pc10_genes <- as.data.frame(TCT_combined_vst[TCT_combined_vst$X1 %in% ensembl_p10,])
pc14_genes <- as.data.frame(TCT_combined_vst[TCT_combined_vst$X1 %in% ensembl_p14,])


for (i in 1:dim(pc10_genes)[1]) {
  pc10_genes[i,2:26] <- zscore(as.numeric(pc10_genes[i,2:26]))
}

for (i in 1:dim(pc14_genes)[1]) {
  pc14_genes[i,2:26] <- zscore(as.numeric(pc14_genes[i,2:26]))
}

heatmap_matrix_pc10 <- as.matrix(pc10_genes[,2:26])
heatmap_matrix_pc14 <- as.matrix(pc14_genes[,2:26])


png('TCT_MKI vs V in cCD PC10 Genes.png',
    width = 800,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc10,
          main = "TCT MKI vs V\nin cCD PC10 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TCT_MKI vs V in cCD PC10 Genes.pdf',
    width = 8,
    height = 10)
heatmap.2(heatmap_matrix_pc10,
          main = "TCT MKI vs V\nin cCD PC10 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TCT_MKI vs V in cCD PC10 Genes.png')

png('TCT_MKI vs V in cCD PC14 Genes.png',
    width = 800,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc14,
          main = "TCT MKI vs V\nin cCD PC14 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TCT_MKI vs V in cCD PC14 Genes.pdf',
    width = 8,
    height = 10)
heatmap.2(heatmap_matrix_pc14,
          main = "TCT MKI vs V\nin cCD PC14 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TCT_MKI vs V in cCD PC14 Genes.png')
```

DEG from PC10 and PC14. The standard of DEG is relaxed to p < 0.1 here to keep consistent with previous analysis and LFC > 0.1.
```{r fig.height = 10, fig.align = "center"}
pc10_dif <- TCT_combined_dif[TCT_combined_dif$X1 %in% ensembl_p10,]
pc14_dif <- TCT_combined_dif[TCT_combined_dif$X1 %in% ensembl_p14,]

pc10_DEG <- subset(pc10_dif, pc10_dif$padj < 0.1 & abs(pc10_dif$log2FoldChange) > 0.1)
pc14_DEG <- subset(pc14_dif, pc14_dif$padj < 0.1 & abs(pc14_dif$log2FoldChange) > 0.1)

pc10_genes_DEG <- pc10_genes[pc10_genes$X1 %in% pc10_DEG$X1,]
pc14_genes_DEG <- pc14_genes[pc14_genes$X1 %in% pc14_DEG$X1,]

pc10_genes_DEG <- inner_join(pc10_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))
pc14_genes_DEG <- inner_join(pc14_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))

print(paste("The number of MK2 inhibitor targets from combined TCT model from cCD PC10:", dim(pc10_genes_DEG)[1]))
print(paste("The number of MK2 inhibitor targets from combined TCT model from cCD PC14:", dim(pc14_genes_DEG)[1]))
```


#### TNF model stratifying based on inflammation status
All genes in the list of 325 from PC10 and PC14
```{r fig.height = 10, fig.align = "center"}

pc10_genes <- as.data.frame(TNF_imki_iv_vst[TNF_imki_iv_vst$X1 %in% ensembl_p10,])
pc14_genes <- as.data.frame(TNF_imki_iv_vst[TNF_imki_iv_vst$X1 %in% ensembl_p14,])


for (i in 1:dim(pc10_genes)[1]) {
  pc10_genes[i,2:12] <- zscore(as.numeric(pc10_genes[i,2:12]))
}

for (i in 1:dim(pc14_genes)[1]) {
  pc14_genes[i,2:12] <- zscore(as.numeric(pc14_genes[i,2:12]))
}

heatmap_matrix_pc10 <- as.matrix(pc10_genes[,2:12])
heatmap_matrix_pc14 <- as.matrix(pc14_genes[,2:12])


png('TNF_I_MKI vs I_V in cCD PC10 Genes.png',
    width = 600,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc10,
          main = "TNF I_MKI vs I_V\nin cCD PC10 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TNF_I_MKI vs I_V in cCD PC10 Genes.pdf',
    width = 6,
    height = 10)
heatmap.2(heatmap_matrix_pc10,
          main = "TNF I_MKI vs I_V\nin cCD PC10 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TNF_I_MKI vs I_V in cCD PC10 Genes.png')

png('TNF_I_MKI vs I_V in cCD PC14 Genes.png',
    width = 600,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc14,
          main = "TNF I_MKI vs I_V\nin cCD PC14 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TNF_I_MKI vs I_V in cCD PC14 Genes.pdf',
    width = 6,
    height = 10)
heatmap.2(heatmap_matrix_pc14,
          main = "TNF I_MKI vs I_V\nin cCD PC14 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TNF_I_MKI vs I_V in cCD PC14 Genes.png')
```

DEG from PC10 and PC14. The standard of DEG is relaxed to p < 0.1 here to keep consistent with previous analysis and LFC > 0.1.
```{r fig.height = 10, fig.align = "center"}
pc10_dif <- TNF_imki_iv_dif[TNF_imki_iv_dif$X1 %in% ensembl_p10,]
pc14_dif <- TNF_imki_iv_dif[TNF_imki_iv_dif$X1 %in% ensembl_p14,]

pc10_DEG <- subset(pc10_dif, pc10_dif$padj < 0.1  & abs(pc10_dif$log2FoldChange > 0.1))
pc14_DEG <- subset(pc14_dif, pc14_dif$padj < 0.1  & abs(pc14_dif$log2FoldChange > 0.1))

pc10_genes_DEG <- pc10_genes[pc10_genes$X1 %in% pc10_DEG$X1,]
pc14_genes_DEG <- pc14_genes[pc14_genes$X1 %in% pc14_DEG$X1,]

pc10_genes_DEG <- inner_join(pc10_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))
pc14_genes_DEG <- inner_join(pc14_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))

print(paste("The number of MK2 inhibitor targets from TNF model from cCD PC10:", dim(pc10_genes_DEG)[1]))
print(paste("The number of MK2 inhibitor targets from TNF model from cCD PC14:", dim(pc14_genes_DEG)[1]))
```

#### TNF model disregarding inflammation status
All genes in the list of 325 from PC10 and PC14
```{r fig.height = 10, fig.align = "center"}

pc10_genes <- as.data.frame(TNF_combined_vst[TNF_combined_vst$X1 %in% ensembl_p10,])
pc14_genes <- as.data.frame(TNF_combined_vst[TNF_combined_vst$X1 %in% ensembl_p14,])


for (i in 1:dim(pc10_genes)[1]) {
  pc10_genes[i,2:22] <- zscore(as.numeric(pc10_genes[i,2:22]))
}

for (i in 1:dim(pc14_genes)[1]) {
  pc14_genes[i,2:22] <- zscore(as.numeric(pc14_genes[i,2:22]))
}

heatmap_matrix_pc10 <- as.matrix(pc10_genes[,2:22])
heatmap_matrix_pc14 <- as.matrix(pc14_genes[,2:22])


png('TNF_MKI vs V in cCD PC10 Genes.png',
    width = 800,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc10,
          main = "TNF MKI vs V\nin cCD PC10 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TNF_MKI vs V in cCD PC10 Genes.pdf',
    width = 8,
    height = 10)
heatmap.2(heatmap_matrix_pc10,
          main = "TNF MKI vs V\nin cCD PC10 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TNF_MKI vs V in cCD PC10 Genes.png')

png('TNF_MKI vs V in cCD PC14 Genes.png',
    width = 800,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc14,
          main = "TNF MKI vs V\nin cCD PC14 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TNF_MKI vs V in cCD PC14 Genes.pdf',
    width = 8,
    height = 10)
heatmap.2(heatmap_matrix_pc14,
          main = "TNF MKI vs V\nin cCD PC14 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TNF_MKI vs V in cCD PC14 Genes.png')
```


DEG from PC10 and PC14. The standard of DEG is relaxed to p < 0.1 here to keep consistent with previous analysis and LFC > 0.1.
```{r fig.height = 10, fig.align = "center"}
pc10_dif <- TNF_combined_dif[TNF_combined_dif$X1 %in% ensembl_p10,]
pc14_dif <- TNF_combined_dif[TNF_combined_dif$X1 %in% ensembl_p14,]

pc10_DEG <- subset(pc10_dif, pc10_dif$padj < 0.1 & abs(pc10_dif$log2FoldChange) > 0.1)
pc14_DEG <- subset(pc14_dif, pc14_dif$padj < 0.1 & abs(pc14_dif$log2FoldChange) > 0.1)

pc10_genes_DEG <- pc10_genes[pc10_genes$X1 %in% pc10_DEG$X1,]
pc14_genes_DEG <- pc14_genes[pc14_genes$X1 %in% pc14_DEG$X1,]

pc10_genes_DEG <- inner_join(pc10_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))
pc14_genes_DEG <- inner_join(pc14_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))

print(paste("The number of MK2 inhibitor targets from TNF model from cCD PC10:", dim(pc10_genes_DEG)[1]))
print(paste("The number of MK2 inhibitor targets from TNF model from cCD PC14:", dim(pc14_genes_DEG)[1]))
```
We cannot make heatmap with only 1 item as input. So I will just print out the item here.
```{r fig.height = 10, fig.align = "center"}
pc10_DEG
```

I would also like to plot the Loading plot for this gene in PC10 and PC14 space.

```{r}
loading_cord <- cCD_PC[cCD_PC$`Ensembl Gene ID` %in% pc10_genes_DEG$X1,]
loading_cord <- loading_cord[, c('cCD_PC10', 'cCD_PC14', 'Ensembl Gene ID')]
loading_cord <- inner_join(loading_cord, ensembl_id[,c(2,4)], by = c("Ensembl Gene ID"="Ensembl Gene ID"))
loading_cord$`Ensembl Gene ID` = NULL
loading_cord[2,] <- loading_cord[1,]
loading_cord <- as.data.frame(loading_cord)
loading_cord[1,] <- c(0,0,NA)
loading_cord$cCD_PC10 <- as.numeric(loading_cord$cCD_PC10)
loading_cord$cCD_PC14 <- as.numeric(loading_cord$cCD_PC14)


ggplot(loading_cord, aes(x=cCD_PC14, y=cCD_PC10)) + 
  geom_line(color = "red",
            arrow = arrow(ends = "first")) +
  xlim(-0.1, 0.1) +
  ylim(-0.1, 0.1) +
  ggtitle("Loading plot for combined TNF MK2i target in cCD PC10 and PC14 space") +
  geom_dl(aes(label = loading_cord$`Marker Symbol`), method = list(dl.trans(x = x + 0.2), "last.points", cex = 0.8))

```

## Heatmap plotting based on iCD PC {#iCDpc}
For iCD PCs, PC1, PC14 and PC16 are selected because they seperate TNF from TCT mice.

The goal of this is to examine how many of the genes that contribute to the seperation of TNF and TCT model based on human PCs are targeted by MK2 inhibitor.

For each PC, the top 325 contributing genes are selected for plotting the heatmap (10% of the list). Top contributing genes are genes with the largest absolute loading coefficients.
```{r fig.height = 10, fig.align = "center"}
threshold_p1 <- quantile(abs(iCD_PC$iCD_PC1), prob=1-n/100)
threshold_p14 <- quantile(abs(iCD_PC$iCD_PC14), prob=1-n/100)
threshold_p16 <- quantile(abs(iCD_PC$iCD_PC16), prob=1-n/100)

ensembl_p1 <- iCD_PC$`Ensembl Gene ID`[abs(iCD_PC$iCD_PC1) > threshold_p1]
ensembl_p14 <- iCD_PC$`Ensembl Gene ID`[abs(iCD_PC$iCD_PC14) > threshold_p14]
ensembl_p16 <- iCD_PC$`Ensembl Gene ID`[abs(iCD_PC$iCD_PC16) > threshold_p16]
```

I_MKI vs I_V datasets will be used here for the analysis.

#### TCT model stratifying based on inflammation status
All genes in the list of 325 from PC1, PC14 and PC16
```{r fig.height = 10, fig.align = "center"}
pc1_genes <- as.data.frame(TCT_imki_iv_vst[TCT_imki_iv_vst$X1 %in% ensembl_p1,])
pc14_genes <- as.data.frame(TCT_imki_iv_vst[TCT_imki_iv_vst$X1 %in% ensembl_p14,])
pc16_genes <-as.data.frame(TCT_imki_iv_vst[TCT_imki_iv_vst$X1 %in% ensembl_p16,])


for (i in 1:dim(pc1_genes)[1]) {
  pc1_genes[i,2:13] <- zscore(as.numeric(pc1_genes[i,2:13]))
}

for (i in 1:dim(pc14_genes)[1]) {
  pc14_genes[i,2:13] <- zscore(as.numeric(pc14_genes[i,2:13]))
}

for (i in 1:dim(pc16_genes)[1]) {
  pc16_genes[i,2:13] <- zscore(as.numeric(pc16_genes[i,2:13]))
}


heatmap_matrix_pc1 <- as.matrix(pc1_genes[,2:13])
heatmap_matrix_pc14 <- as.matrix(pc14_genes[,2:13])
heatmap_matrix_pc16 <- as.matrix(pc16_genes[,2:13])


png('TCT_I_MKI vs I_V in iCD PC1 Genes.png',
    width = 600,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc1,
          main = "TCT I_MKI vs I_V\nin iCD PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TCT_I_MKI vs I_V in iCD PC1 Genes.pdf',
    width = 6,
    height = 10)
heatmap.2(heatmap_matrix_pc1,
          main = "TCT I_MKI vs I_V\nin iCD PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TCT_I_MKI vs I_V in iCD PC1 Genes.png')

png('TCT_I_MKI vs I_V in iCD PC14 Genes.png',
    width = 600,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc14,
          main = "TCT I_MKI vs I_V\nin iCD PC14 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TCT_I_MKI vs I_V in iCD PC14 Genes.pdf',
    width = 6,
    height = 10)
heatmap.2(heatmap_matrix_pc14,
          main = "TCT I_MKI vs I_V\nin iCD PC14 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TCT_I_MKI vs I_V in iCD PC14 Genes.png')


png('TCT_I_MKI vs I_V in iCD PC16 Genes.png',
    width = 600,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc16,
          main = "TCT I_MKI vs I_V\nin iCD PC16 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TCT_I_MKI vs I_V in iCD PC16 Genes.pdf',
    width = 6,
    height = 10)
heatmap.2(heatmap_matrix_pc16,
          main = "TCT I_MKI vs I_V\nin iCD PC16 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TCT_I_MKI vs I_V in iCD PC16 Genes.png')

```

DEG from PC1, PC14 and PC16. The standard of DEG is relaxed to p < 0.1 here to keep consistent with previous analysis and LFC > 0.1.
```{r fig.height = 10, fig.align = "center"}
pc1_dif <- TCT_imki_iv_dif[TCT_imki_iv_dif$X1 %in% ensembl_p1,]
pc14_dif <- TCT_imki_iv_dif[TCT_imki_iv_dif$X1 %in% ensembl_p14,]
pc16_dif <- TCT_imki_iv_dif[TCT_imki_iv_dif$X1 %in% ensembl_p16,]


pc1_DEG <- subset(pc1_dif, pc1_dif$padj < 0.1  & abs(pc1_dif$log2FoldChange > 0.1))
pc14_DEG <- subset(pc14_dif, pc14_dif$padj < 0.1  & abs(pc14_dif$log2FoldChange > 0.1))
pc16_DEG <- subset(pc16_dif, pc16_dif$padj < 0.1  & abs(pc16_dif$log2FoldChange > 0.1))


pc1_genes_DEG <- pc1_genes[pc1_genes$X1 %in% pc1_DEG$X1,]
pc14_genes_DEG <- pc14_genes[pc14_genes$X1 %in% pc14_DEG$X1,]
pc16_genes_DEG <- pc16_genes[pc16_genes$X1 %in% pc16_DEG$X1,]


pc1_genes_DEG <- inner_join(pc1_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))
pc14_genes_DEG <- inner_join(pc14_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))
pc16_genes_DEG <- inner_join(pc16_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))


print(paste("The number of MK2 inhibitor targets from TCT model from iCD PC1:", dim(pc1_genes_DEG)[1]))
print(paste("The number of MK2 inhibitor targets from TCT model from iCD PC14:", dim(pc14_genes_DEG)[1]))
print(paste("The number of MK2 inhibitor targets from TCT model from iCD PC16:", dim(pc16_genes_DEG)[1]))

```

#### TCT model disregarding inflammation status
All genes in the list of 325 from PC1, PC14 and PC16
```{r fig.height = 10, fig.align = "center"}
pc1_genes <- as.data.frame(TCT_combined_vst[TCT_combined_vst$X1 %in% ensembl_p1,])
pc14_genes <- as.data.frame(TCT_combined_vst[TCT_combined_vst$X1 %in% ensembl_p14,])
pc16_genes <-as.data.frame(TCT_combined_vst[TCT_combined_vst$X1 %in% ensembl_p16,])


for (i in 1:dim(pc1_genes)[1]) {
  pc1_genes[i,2:26] <- zscore(as.numeric(pc1_genes[i,2:26]))
}

for (i in 1:dim(pc14_genes)[1]) {
  pc14_genes[i,2:26] <- zscore(as.numeric(pc14_genes[i,2:26]))
}

for (i in 1:dim(pc16_genes)[1]) {
  pc16_genes[i,2:26] <- zscore(as.numeric(pc16_genes[i,2:26]))
}


heatmap_matrix_pc1 <- as.matrix(pc1_genes[,2:26])
heatmap_matrix_pc14 <- as.matrix(pc14_genes[,2:26])
heatmap_matrix_pc16 <- as.matrix(pc16_genes[,2:26])


png('TCT_MKI vs V in iCD PC1 Genes.png',
    width = 800,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc1,
          main = "TCT MKI vs V\nin iCD PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TCT_MKI vs V in iCD PC1 Genes.pdf',
    width = 8,
    height = 10)
heatmap.2(heatmap_matrix_pc1,
          main = "TCT MKI vs V\nin iCD PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics("TCT_MKI vs V in iCD PC1 Genes.png")

png('TCT_MKI vs V in iCD PC14 Genes.png',
    width = 800,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc14,
          main = "TCT MKI vs V\nin iCD PC14 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TCT_MKI vs V in iCD PC14 Genes.pdf',
    width = 8,
    height = 10)
heatmap.2(heatmap_matrix_pc14,
          main = "TCT MKI vs V\nin iCD PC14 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics("TCT_MKI vs V in iCD PC14 Genes.png")


png('TCT_MKI vs V in iCD PC16 Genes.png',
    width = 800,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc16,
          main = "TCT MKI vs V\nin iCD PC16 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TCT_MKI vs V in iCD PC16 Genes.pdf',
    width = 8,
    height = 10)
heatmap.2(heatmap_matrix_pc16,
          main = "TCT MKI vs V\nin iCD PC16 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics("TCT_MKI vs V in iCD PC16 Genes.png")

```

DEG from PC1, PC14 and PC16. The standard of DEG is relaxed to p < 0.1 here to keep consistent with previous analysis and LFC > 0.1.
```{r fig.height = 10, fig.align = "center"}
pc1_dif <- TCT_combined_dif[TCT_combined_dif$X1 %in% ensembl_p1,]
pc14_dif <- TCT_combined_dif[TCT_combined_dif$X1 %in% ensembl_p14,]
pc16_dif <- TCT_combined_dif[TCT_combined_dif$X1 %in% ensembl_p16,]


pc1_DEG <- subset(pc1_dif, pc1_dif$padj < 0.1  & abs(pc1_dif$log2FoldChange > 0.1))
pc14_DEG <- subset(pc14_dif, pc14_dif$padj < 0.1  & abs(pc14_dif$log2FoldChange > 0.1))
pc16_DEG <- subset(pc16_dif, pc16_dif$padj < 0.1  & abs(pc16_dif$log2FoldChange > 0.1))


pc1_genes_DEG <- pc1_genes[pc1_genes$X1 %in% pc1_DEG$X1,]
pc14_genes_DEG <- pc14_genes[pc14_genes$X1 %in% pc14_DEG$X1,]
pc16_genes_DEG <- pc16_genes[pc16_genes$X1 %in% pc16_DEG$X1,]


pc1_genes_DEG <- inner_join(pc1_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))
pc14_genes_DEG <- inner_join(pc14_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))
pc16_genes_DEG <- inner_join(pc16_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))


print(paste("The number of MK2 inhibitor targets from TCT model from iCD PC1:", dim(pc1_genes_DEG)[1]))
print(paste("The number of MK2 inhibitor targets from TCT model from iCD PC14:", dim(pc14_genes_DEG)[1]))
print(paste("The number of MK2 inhibitor targets from TCT model from iCD PC16:", dim(pc16_genes_DEG)[1]))

```

#### TNF model stratifying based on inflammation status
All genes in the list of 325 from PC1, PC14 and PC16
```{r fig.height = 10, fig.align = "center"}
pc1_genes <- as.data.frame(TNF_imki_iv_vst[TNF_imki_iv_vst$X1 %in% ensembl_p1,])
pc14_genes <- as.data.frame(TNF_imki_iv_vst[TNF_imki_iv_vst$X1 %in% ensembl_p14,])
pc16_genes <- as.data.frame(TNF_imki_iv_vst[TNF_imki_iv_vst$X1 %in% ensembl_p16,])


for (i in 1:dim(pc1_genes)[1]) {
  pc1_genes[i,2:12] <- zscore(as.numeric(pc1_genes[i,2:12]))
}

for (i in 1:dim(pc14_genes)[1]) {
  pc14_genes[i,2:12] <- zscore(as.numeric(pc14_genes[i,2:12]))
}

for (i in 1:dim(pc16_genes)[1]) {
  pc16_genes[i,2:12] <- zscore(as.numeric(pc16_genes[i,2:12]))
}


heatmap_matrix_pc1 <- as.matrix(pc1_genes[,2:12])
heatmap_matrix_pc14 <- as.matrix(pc14_genes[,2:12])
heatmap_matrix_pc16 <- as.matrix(pc16_genes[,2:12])



png('TNF_I_MKI vs I_V in iCD PC1 Genes.png',
    width = 600,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc1,
          main = "TNF I_MKI vs I_V\nin iCD PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TNF_I_MKI vs I_V in iCD PC1 Genes.pdf',
    width = 6,
    height = 10)
heatmap.2(heatmap_matrix_pc1,
          main = "TNF I_MKI vs I_V\nin iCD PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TNF_I_MKI vs I_V in iCD PC1 Genes.png')

png('TNF_I_MKI vs I_V in iCD PC14 Genes.png',
    width = 600,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc14,
          main = "TNF I_MKI vs I_V\nin iCD PC14 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TNF_I_MKI vs I_V in iCD PC14 Genes.pdf',
    width = 6,
    height = 10)
heatmap.2(heatmap_matrix_pc14,
          main = "TNF I_MKI vs I_V\nin iCD PC14 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TNF_I_MKI vs I_V in iCD PC14 Genes.png')

png('TNF_I_MKI vs I_V in iCD PC16 Genes.png',
    width = 600,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc16,
          main = "TNF I_MKI vs I_V\nin iCD PC16 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TNF_I_MKI vs I_V in iCD PC16 Genes.pdf',
    width = 6,
    height = 10)
heatmap.2(heatmap_matrix_pc16,
          main = "TNF I_MKI vs I_V\nin iCD PC16 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TNF_I_MKI vs I_V in iCD PC16 Genes.png')

```


DEG from PC1, PC14 and PC16. The standard of DEG is relaxed to p < 0.1 here to keep consistent with previous analysis and LFC > 0.1.
```{r fig.height = 10, fig.align = "center"}
pc1_dif <- TNF_imki_iv_dif[TNF_imki_iv_dif$X1 %in% ensembl_p1,]
pc14_dif <- TNF_imki_iv_dif[TNF_imki_iv_dif$X1 %in% ensembl_p14,]
pc16_dif <- TNF_imki_iv_dif[TNF_imki_iv_dif$X1 %in% ensembl_p16,]

pc1_DEG <- subset(pc1_dif, pc1_dif$padj < 0.1  & abs(pc1_dif$log2FoldChange > 0.1))
pc14_DEG <- subset(pc14_dif, pc14_dif$padj < 0.1  & abs(pc14_dif$log2FoldChange > 0.1))
pc16_DEG <- subset(pc16_dif, pc16_dif$padj < 0.1  & abs(pc16_dif$log2FoldChange > 0.1))

pc1_genes_DEG <- pc1_genes[pc1_genes$X1 %in% pc1_DEG$X1,]
pc14_genes_DEG <- pc14_genes[pc14_genes$X1 %in% pc14_DEG$X1,]
pc16_genes_DEG <- pc16_genes[pc16_genes$X1 %in% pc16_DEG$X1,]


pc1_genes_DEG <- inner_join(pc1_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))
pc14_genes_DEG <- inner_join(pc14_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))
pc16_genes_DEG <- inner_join(pc16_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))


print(paste("The number of MK2 inhibitor targets from TNF model from iCD PC1:", dim(pc1_genes_DEG)[1]))
print(paste("The number of MK2 inhibitor targets from TNF model from iCD PC14:", dim(pc14_genes_DEG)[1]))
print(paste("The number of MK2 inhibitor targets from TNF model from iCD PC16:", dim(pc16_genes_DEG)[1]))

```

Let's take a look at the only MK2i target from TNF model in iCD PC1 space.
```{r fig.height = 10, fig.align = "center"}
pc1_DEG
```
I would also like to plot the Loading plot for this gene in PC1/PC14 space and PC1/PC16 space.

```{r}
loading_cord <- iCD_PC[iCD_PC$`Ensembl Gene ID` %in% pc1_genes_DEG$X1,]
loading_cord <- loading_cord[, c('iCD_PC1', 'iCD_PC14', 'iCD_PC16', 'Ensembl Gene ID')]
loading_cord <- inner_join(loading_cord, ensembl_id[,c(2,4)], by = c("Ensembl Gene ID"="Ensembl Gene ID"))
loading_cord$`Ensembl Gene ID` = NULL
loading_cord[2,] <- loading_cord[1,]
loading_cord <- as.data.frame(loading_cord)
loading_cord[1,] <- c(0,0,0,NA)
loading_cord$iCD_PC1 <- as.numeric(loading_cord$iCD_PC1)
loading_cord$iCD_PC14 <- as.numeric(loading_cord$iCD_PC14)
loading_cord$iCD_PC16 <- as.numeric(loading_cord$iCD_PC16)

ggplot(loading_cord, aes(x=iCD_PC1, y=iCD_PC14)) + 
  geom_line(color = "red",
            arrow = arrow(ends = "first")) +
  xlim(-0.1, 0.1) +
  ylim(-0.1, 0.1) +
  ggtitle("Loading plot for TNF MK2i target in iCD PC1 and PC14 space") +
  geom_dl(aes(label = loading_cord$`Marker Symbol`), method = list(dl.trans(x = x - 1), dl.trans(y = y - 0.5), "last.points", cex = 0.8))

ggplot(loading_cord, aes(x=iCD_PC1, y=iCD_PC16)) + 
  geom_line(color = "red",
            arrow = arrow(ends = "first")) +
  xlim(-0.1, 0.1) +
  ylim(-0.1, 0.1) +
  ggtitle("Loading plot for TNF MK2i target in iCD PC1 and PC16 space") +
  geom_dl(aes(label = loading_cord$`Marker Symbol`), method = list(dl.trans(x = x - 1), dl.trans(y = y - 0.5), "last.points", cex = 0.8))

```

#### TNF model disregarding inflammation status
All genes in the list of 325 from PC1, PC14 and PC16
```{r fig.height = 10, fig.align = "center"}
pc1_genes <- as.data.frame(TNF_combined_vst[TNF_combined_vst$X1 %in% ensembl_p1,])
pc14_genes <- as.data.frame(TNF_combined_vst[TNF_combined_vst$X1 %in% ensembl_p14,])
pc16_genes <-as.data.frame(TNF_combined_vst[TNF_combined_vst$X1 %in% ensembl_p16,])


for (i in 1:dim(pc1_genes)[1]) {
  pc1_genes[i,2:22] <- zscore(as.numeric(pc1_genes[i,2:22]))
}

for (i in 1:dim(pc14_genes)[1]) {
  pc14_genes[i,2:22] <- zscore(as.numeric(pc14_genes[i,2:22]))
}

for (i in 1:dim(pc16_genes)[1]) {
  pc16_genes[i,2:22] <- zscore(as.numeric(pc16_genes[i,2:22]))
}


heatmap_matrix_pc1 <- as.matrix(pc1_genes[,2:22])
heatmap_matrix_pc14 <- as.matrix(pc14_genes[,2:22])
heatmap_matrix_pc16 <- as.matrix(pc16_genes[,2:22])


png('TNF_MKI vs V in iCD PC1 Genes.png',
    width = 800,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc1,
          main = "TNF MKI vs V\nin iCD PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TNF_MKI vs V in iCD PC1 Genes.pdf',
    width = 8,
    height = 10)
heatmap.2(heatmap_matrix_pc1,
          main = "TNF MKI vs V\nin iCD PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TNF_MKI vs V in iCD PC1 Genes.png')

png('TNF_MKI vs V in iCD PC14 Genes.png',
    width = 800,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc14,
          main = "TNF MKI vs V\nin iCD PC14 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TNF_MKI vs V in iCD PC14 Genes.pdf',
    width = 8,
    height = 10)
heatmap.2(heatmap_matrix_pc14,
          main = "TNF MKI vs V\nin iCD PC14 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TNF_MKI vs V in iCD PC14 Genes.png')


png('TNF_MKI vs V in iCD PC16 Genes.png',
    width = 800,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc16,
          main = "TNF MKI vs V\nin iCD PC16 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TNF_MKI vs V in iCD PC16 Genes.pdf',
    width = 8,
    height = 10)
heatmap.2(heatmap_matrix_pc16,
          main = "TNF MKI vs V\nin iCD PC16 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TNF_MKI vs V in iCD PC16 Genes.png')

```

DEG from PC1, PC14 and PC16. The standard of DEG is relaxed to p < 0.1 here to keep consistent with previous analysis and LFC > 0.1.
```{r fig.height = 10, fig.align = "center"}
pc1_dif <- TNF_combined_dif[TNF_combined_dif$X1 %in% ensembl_p1,]
pc14_dif <- TNF_combined_dif[TNF_combined_dif$X1 %in% ensembl_p14,]
pc16_dif <- TNF_combined_dif[TNF_combined_dif$X1 %in% ensembl_p16,]


pc1_DEG <- subset(pc1_dif, pc1_dif$padj < 0.1  & abs(pc1_dif$log2FoldChange > 0.1))
pc14_DEG <- subset(pc14_dif, pc14_dif$padj < 0.1  & abs(pc14_dif$log2FoldChange > 0.1))
pc16_DEG <- subset(pc16_dif, pc16_dif$padj < 0.1  & abs(pc16_dif$log2FoldChange > 0.1))


pc1_genes_DEG <- pc1_genes[pc1_genes$X1 %in% pc1_DEG$X1,]
pc14_genes_DEG <- pc14_genes[pc14_genes$X1 %in% pc14_DEG$X1,]
pc16_genes_DEG <- pc16_genes[pc16_genes$X1 %in% pc16_DEG$X1,]


pc1_genes_DEG <- inner_join(pc1_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))
pc14_genes_DEG <- inner_join(pc14_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))
pc16_genes_DEG <- inner_join(pc16_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))


print(paste("The number of MK2 inhibitor targets from TNF model from iCD PC1:", dim(pc1_genes_DEG)[1]))
print(paste("The number of MK2 inhibitor targets from TNF model from iCD PC14:", dim(pc14_genes_DEG)[1]))
print(paste("The number of MK2 inhibitor targets from TNF model from iCD PC16:", dim(pc16_genes_DEG)[1]))

```

Let's take a look at the only MK2i target from TNF model in iCD PC1 space.
```{r fig.height = 10, fig.align = "center"}
pc14_DEG
```
I would also like to plot the Loading plot for this gene in PC1/PC14 space and PC1/PC16 space.

```{r}
loading_cord <- iCD_PC[iCD_PC$`Ensembl Gene ID` %in% pc14_genes_DEG$X1,]
loading_cord <- loading_cord[, c('iCD_PC1', 'iCD_PC14', 'iCD_PC16', 'Ensembl Gene ID')]
loading_cord <- inner_join(loading_cord, ensembl_id[,c(2,4)], by = c("Ensembl Gene ID"="Ensembl Gene ID"))
loading_cord$`Ensembl Gene ID` = NULL
loading_cord[2,] <- loading_cord[1,]
loading_cord <- as.data.frame(loading_cord)
loading_cord[1,] <- c(0,0,0,NA)
loading_cord$iCD_PC1 <- as.numeric(loading_cord$iCD_PC1)
loading_cord$iCD_PC14 <- as.numeric(loading_cord$iCD_PC14)
loading_cord$iCD_PC16 <- as.numeric(loading_cord$iCD_PC16)

ggplot(loading_cord, aes(x=iCD_PC1, y=iCD_PC14)) + 
  geom_line(color = "red",
            arrow = arrow(ends = "first")) +
  xlim(-0.1, 0.1) +
  ylim(-0.1, 0.1) +
  ggtitle("Loading plot for TNF MK2i target in iCD PC1 and PC14 space") +
  geom_dl(aes(label = loading_cord$`Marker Symbol`), method = list(dl.trans(x = x - 1), dl.trans(y = y - 0.5), "last.points", cex = 0.8))

ggplot(loading_cord, aes(x=iCD_PC1, y=iCD_PC16)) + 
  geom_line(color = "red",
            arrow = arrow(ends = "first")) +
  xlim(-0.1, 0.1) +
  ylim(-0.1, 0.1) +
  ggtitle("Loading plot for TNF MK2i target in iCD PC1 and PC16 space") +
  geom_dl(aes(label = loading_cord$`Marker Symbol`), method = list(dl.trans(x = x - 1), dl.trans(y = y - 0.5), "last.points", cex = 0.8))

```


## Heatmap plotting based on UC PC using MK2i dataset {#UCpc}
TCT and TNF MK2i treated dataset with and without inflammation are used here.

```{r fig.height = 10, fig.align = "center"}
# ID conversion
# load the PC dataset
UC_PC <- read_csv("UC PC.csv")

human_id <- as.data.frame(UC_PC$Row)
human_id <- as_tibble(human_id)
colnames(human_id) <- "human_gene_symbol"

id_conversion <- inner_join(human_id, homolog_id, by = c("human_gene_symbol" = "Human Marker Symbol"))
non_duplicates_idx <- which(duplicated(id_conversion$human_gene_symbol) == FALSE)
id_conversion <- id_conversion[non_duplicates_idx, ]

id_conversion <- inner_join(id_conversion, ensembl_id, by = c("Mouse Marker Symbol" = "Marker Symbol"))
non_duplicates_idx <- which(duplicated(id_conversion$`Mouse Marker Symbol`) == FALSE)
id_conversion <- id_conversion[non_duplicates_idx, ]

id_conversion <- id_conversion[,c(1,7)]

UC_PC <- inner_join(UC_PC, id_conversion, by = c("Row" = "human_gene_symbol"))
```
For UC PCs, PC1 and PC2 are selected because they seperates MK2i treatment statuses in TCT mice. 

The goal of this is to examine how many of the genes that contribute to the seperation of MK2i vs Vehicle treated TCT mice based on human PCs are targeted by MK2 inhibitor.

For each PC, the top 49 contributing genes are selected for plotting the heatmap (10% of the list). Top contributing genes are genes with the largest absolute loading coefficients.
```{r fig.height = 10, fig.align = "center"}
n = 10
threshold_p1 <- quantile(abs(UC_PC$Var1_1), prob=1-n/100)
threshold_p2 <- quantile(abs(UC_PC$Var1_2), prob=1-n/100)

ensembl_p1 <- UC_PC$`Ensembl Gene ID`[abs(UC_PC$Var1_1) > threshold_p1]
ensembl_p2 <- UC_PC$`Ensembl Gene ID`[abs(UC_PC$Var1_2) > threshold_p2]
```

#### TCT model with inflammation treated with MK2i 
All genes in the list of 49 from PC1 and PC2
```{r fig.height = 10, fig.align = "center"}

pc1_genes <- as.data.frame(TCT_imki_iv_vst[TCT_imki_iv_vst$X1 %in% ensembl_p1,])
pc2_genes <- as.data.frame(TCT_imki_iv_vst[TCT_imki_iv_vst$X1 %in% ensembl_p2,])


for (i in 1:dim(pc1_genes)[1]) {
  pc1_genes[i,2:13] <- zscore(as.numeric(pc1_genes[i,2:13]))
}

for (i in 1:dim(pc2_genes)[1]) {
  pc2_genes[i,2:13] <- zscore(as.numeric(pc2_genes[i,2:13]))
}

heatmap_matrix_pc1 <- as.matrix(pc1_genes[,2:13])
heatmap_matrix_pc2 <- as.matrix(pc2_genes[,2:13])


png('TCT_I_MKI vs I_V in UC PC1 Genes.png',
    width = 600,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc1,
          main = "TCT I_MKI vs I_V\nin UC PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TCT_I_MKI vs I_V in UC PC1 Genes.pdf',
    width = 6,
    height = 10)
heatmap.2(heatmap_matrix_pc1,
          main = "TCT I_MKI vs I_V\nin UC PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TCT_I_MKI vs I_V in UC PC1 Genes.png')

png('TCT_I_MKI vs I_V in UC PC2 Genes.png',
    width = 600,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc2,
          main = "TCT I_MKI vs I_V\nin UC PC2 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TCT_I_MKI vs I_V in UC PC2 Genes.pdf',
    width = 6,
    height = 10)
heatmap.2(heatmap_matrix_pc2,
          main = "TCT I_MKI vs I_V\nin UC PC2 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TCT_I_MKI vs I_V in UC PC2 Genes.png')

```


DEG from PC1 and PC2. The standard of DEG is relaxed to p < 0.1 here to keep consistent with previous analysis and LFC > 0.1.
```{r fig.height = 10, fig.align = "center"}
pc1_dif <- TCT_imki_iv_dif[TCT_imki_iv_dif$X1 %in% ensembl_p1,]
pc2_dif <- TCT_imki_iv_dif[TCT_imki_iv_dif$X1 %in% ensembl_p2,]
pc1_dif
pc2_dif

pc1_DEG <- subset(pc1_dif, pc1_dif$padj < 0.1  & abs(pc1_dif$log2FoldChange > 0.1))
pc2_DEG <- subset(pc2_dif, pc2_dif$padj < 0.1  & abs(pc2_dif$log2FoldChange > 0.1))

pc1_genes_DEG <- pc1_genes[pc1_genes$X1 %in% pc1_DEG$X1,]
pc2_genes_DEG <- pc2_genes[pc2_genes$X1 %in% pc2_DEG$X1,]

pc1_genes_DEG <- inner_join(pc1_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))
pc2_genes_DEG <- inner_join(pc2_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))

print(paste("The number of MK2 inhibitor targets from TCT model from UC PC1:", dim(pc1_genes_DEG)[1]))
print(paste("The number of MK2 inhibitor targets from TCT model from UC PC2:", dim(pc2_genes_DEG)[1]))
```

#### TCT model with no inflammation treated with MK2i 
All genes in the list of 49 from PC1 and PC2
```{r fig.height = 10, fig.align = "center"}

pc1_genes <- as.data.frame(TCT_nimki_niv_vst[TCT_nimki_niv_vst$X1 %in% ensembl_p1,])
pc2_genes <- as.data.frame(TCT_nimki_niv_vst[TCT_nimki_niv_vst$X1 %in% ensembl_p2,])


for (i in 1:dim(pc1_genes)[1]) {
  pc1_genes[i,2:12] <- zscore(as.numeric(pc1_genes[i,2:12]))
}

for (i in 1:dim(pc2_genes)[1]) {
  pc2_genes[i,2:12] <- zscore(as.numeric(pc2_genes[i,2:12]))
}

heatmap_matrix_pc1 <- as.matrix(pc1_genes[,2:12])
heatmap_matrix_pc2 <- as.matrix(pc2_genes[,2:12])


png('TCT_NI_MKI vs NI_V in UC PC1 Genes.png',
    width = 600,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc1,
          main = "TCT NI_MKI vs NI_V\nin UC PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TCT_NI_MKI vs NI_V in UC PC1 Genes.pdf',
    width = 6,
    height = 10)
heatmap.2(heatmap_matrix_pc1,
          main = "TCT NI_MKI vs NI_V\nin UC PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TCT_NI_MKI vs NI_V in UC PC1 Genes.png')

png('TCT_NI_MKI vs NI_V in UC PC2 Genes.png',
    width = 600,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc2,
          main = "TCT NI_MKI vs NI_V\nin UC PC2 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TCT_NI_MKI vs NI_V in UC PC2 Genes.pdf',
    width = 6,
    height = 10)
heatmap.2(heatmap_matrix_pc2,
          main = "TCT NI_MKI vs NI_V\nin UC PC2 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TCT_NI_MKI vs NI_V in UC PC2 Genes.png')

```

DEG from PC1 and PC2. The standard of DEG is relaxed to p < 0.1 here to keep consistent with previous analysis and LFC > 0.1.
```{r fig.height = 10, fig.align = "center"}
pc1_dif <- TCT_nimki_niv_dif[TCT_nimki_niv_dif$X1 %in% ensembl_p1,]
pc2_dif <- TCT_nimki_niv_dif[TCT_nimki_niv_dif$X1 %in% ensembl_p2,]
pc1_dif
pc2_dif

pc1_DEG <- subset(pc1_dif, pc1_dif$padj < 0.1  & abs(pc1_dif$log2FoldChange > 0.1))
pc2_DEG <- subset(pc2_dif, pc2_dif$padj < 0.1  & abs(pc2_dif$log2FoldChange > 0.1))

pc1_genes_DEG <- pc1_genes[pc1_genes$X1 %in% pc1_DEG$X1,]
pc2_genes_DEG <- pc2_genes[pc2_genes$X1 %in% pc2_DEG$X1,]

pc1_genes_DEG <- inner_join(pc1_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))
pc2_genes_DEG <- inner_join(pc2_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))

print(paste("The number of MK2 inhibitor targets from TCT model from UC PC1:", dim(pc1_genes_DEG)[1]))
print(paste("The number of MK2 inhibitor targets from TCT model from UC PC2:", dim(pc2_genes_DEG)[1]))
```


#### TNF model with inflammation treated with MK2i 
All genes in the list of 49 from PC1 and PC2
```{r fig.height = 10, fig.align = "center"}

pc1_genes <- as.data.frame(TNF_imki_iv_vst[TNF_imki_iv_vst$X1 %in% ensembl_p1,])
pc2_genes <- as.data.frame(TNF_imki_iv_vst[TNF_imki_iv_vst$X1 %in% ensembl_p2,])


for (i in 1:dim(pc1_genes)[1]) {
  pc1_genes[i,2:12] <- zscore(as.numeric(pc1_genes[i,2:12]))
}

for (i in 1:dim(pc2_genes)[1]) {
  pc2_genes[i,2:12] <- zscore(as.numeric(pc2_genes[i,2:12]))
}

heatmap_matrix_pc1 <- as.matrix(pc1_genes[,2:12])
heatmap_matrix_pc2 <- as.matrix(pc2_genes[,2:12])


png('TNF_I_MKI vs I_V in UC PC1 Genes.png',
    width = 600,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc1,
          main = "TNF I_MKI vs I_V\nin UC PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TNF_I_MKI vs I_V in UC PC1 Genes.pdf',
    width = 6,
    height = 10)
heatmap.2(heatmap_matrix_pc1,
          main = "TNF I_MKI vs I_V\nin UC PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TNF_I_MKI vs I_V in UC PC1 Genes.png')

png('TNF_I_MKI vs I_V in UC PC2 Genes.png',
    width = 600,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc2,
          main = "TNF I_MKI vs I_V\nin UC PC2 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TNF_I_MKI vs I_V in UC PC2 Genes.pdf',
    width = 6,
    height = 10)
heatmap.2(heatmap_matrix_pc2,
          main = "TNF I_MKI vs I_V\nin UC PC2 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TNF_I_MKI vs I_V in UC PC2 Genes.png')

```


DEG from PC1 and PC2. The standard of DEG is relaxed to p < 0.1 here to keep consistent with previous analysis and LFC > 0.1.
```{r fig.height = 10, fig.align = "center"}
pc1_dif <- TNF_imki_iv_dif[TNF_imki_iv_dif$X1 %in% ensembl_p1,]
pc2_dif <- TNF_imki_iv_dif[TNF_imki_iv_dif$X1 %in% ensembl_p2,]
pc1_dif
pc2_dif

pc1_DEG <- subset(pc1_dif, pc1_dif$padj < 0.1  & abs(pc1_dif$log2FoldChange > 0.1))
pc2_DEG <- subset(pc2_dif, pc2_dif$padj < 0.1  & abs(pc2_dif$log2FoldChange > 0.1))

pc1_genes_DEG <- pc1_genes[pc1_genes$X1 %in% pc1_DEG$X1,]
pc2_genes_DEG <- pc2_genes[pc2_genes$X1 %in% pc2_DEG$X1,]

pc1_genes_DEG <- inner_join(pc1_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))
pc2_genes_DEG <- inner_join(pc2_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))

print(paste("The number of MK2 inhibitor targets from TNF model from UC PC1:", dim(pc1_genes_DEG)[1]))
print(paste("The number of MK2 inhibitor targets from TNF model from UC PC2:", dim(pc2_genes_DEG)[1]))
```

#### TNF model with no inflammation treated with MK2i 
All genes in the list of 49 from PC1 and PC2
```{r fig.height = 10, fig.align = "center"}

pc1_genes <- as.data.frame(TNF_nimki_niv_vst[TNF_nimki_niv_vst$X1 %in% ensembl_p1,])
pc2_genes <- as.data.frame(TNF_nimki_niv_vst[TNF_nimki_niv_vst$X1 %in% ensembl_p2,])


for (i in 1:dim(pc1_genes)[1]) {
  pc1_genes[i,2:11] <- zscore(as.numeric(pc1_genes[i,2:11]))
}

for (i in 1:dim(pc2_genes)[1]) {
  pc2_genes[i,2:11] <- zscore(as.numeric(pc2_genes[i,2:11]))
}

heatmap_matrix_pc1 <- as.matrix(pc1_genes[,2:11])
heatmap_matrix_pc2 <- as.matrix(pc2_genes[,2:11])


png('TNF_NI_MKI vs NI_V in UC PC1 Genes.png',
    width = 600,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc1,
          main = "TNF NI_MKI vs NI_V\nin UC PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TNF_NI_MKI vs NI_V in UC PC1 Genes.pdf',
    width = 6,
    height = 10)
heatmap.2(heatmap_matrix_pc1,
          main = "TNF NI_MKI vs NI_V\nin UC PC1 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TNF_NI_MKI vs NI_V in UC PC1 Genes.png')

png('TNF_NI_MKI vs NI_V in UC PC2 Genes.png',
    width = 600,
    height = 1200,
    res = 200,
    pointsize = 8)
heatmap.2(heatmap_matrix_pc2,
          main = "TNF NI_MKI vs NI_V\nin UC PC2 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

pdf('TNF_NI_MKI vs NI_V in UC PC2 Genes.pdf',
    width = 6,
    height = 10)
heatmap.2(heatmap_matrix_pc2,
          main = "TNF NI_MKI vs NI_V\nin UC PC2 Genes",
          density.info = "none",
          key = TRUE,
          lhei = c(1,7),
          col=my_palette,
          cexCol = 1,
          margins = c(8,2),
          trace = "none",
          dendrogram = "row",
          labRow = FALSE,
          keysize = 2,
          ylab = "Genes",
          Colv = "NA")
dev.off()

include_graphics('TNF_NI_MKI vs NI_V in UC PC2 Genes.png')

```

DEG from PC1 and PC2. The standard of DEG is relaxed to p < 0.1 here to keep consistent with previous analysis and LFC > 0.1.
```{r fig.height = 10, fig.align = "center"}
pc1_dif <- TNF_nimki_niv_dif[TNF_nimki_niv_dif$X1 %in% ensembl_p1,]
pc2_dif <- TNF_nimki_niv_dif[TNF_nimki_niv_dif$X1 %in% ensembl_p2,]
pc1_dif
pc2_dif

pc1_DEG <- subset(pc1_dif, pc1_dif$padj < 0.1  & abs(pc1_dif$log2FoldChange > 0.1))
pc2_DEG <- subset(pc2_dif, pc2_dif$padj < 0.1  & abs(pc2_dif$log2FoldChange > 0.1))

pc1_genes_DEG <- pc1_genes[pc1_genes$X1 %in% pc1_DEG$X1,]
pc2_genes_DEG <- pc2_genes[pc2_genes$X1 %in% pc2_DEG$X1,]

pc1_genes_DEG <- inner_join(pc1_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))
pc2_genes_DEG <- inner_join(pc2_genes_DEG, ensembl_id[,c(2,4)], by = c("X1" = "Ensembl Gene ID"))

print(paste("The number of MK2 inhibitor targets from TNF model from UC PC1:", dim(pc1_genes_DEG)[1]))
print(paste("The number of MK2 inhibitor targets from TNF model from UC PC2:", dim(pc2_genes_DEG)[1]))
```
